### ISSUE TITLE
MethodParser.fromJson() : "methodName" inside of payload throws exception

### ISSUE THREAD

**Issue by cyansprite** [Azure/azure-iot-sdk-java#754]
# Environment
- Ubuntu 18.04

- openjdk version "1.8.0_242"
  OpenJDK Runtime Environment (build 1.8.0_242-8u242-b08-0ubuntu3~16.04-b08)
  OpenJDK 64-Bit Server VM (build 25.242-b08, mixed mode)

- com.microsoft.azure.sdk.iot:iot-service-client:1.13.0

# Description of the issue:
```java
/* Codes_SRS_METHODPARSER_21_008: [If the provided json is null, empty, or not valid, the fromJson shall throws IllegalArgumentException.] */
throw new IllegalArgumentException("Invoke method name and Status reported in the same json");
```

# Code sample exhibiting the issue:
```java
MethodResult result = methodClient.invoke(deviceId, "CALL_APPLICATION", (long)30, (long)30, payload);
```
Response from direct method:
```json
{
  "status": 200,
  "payload": {
    "methodName": "CALL_APPLICATION",
    "result": 1,
    "status": "COMPLETED"
  }
}
```

**Comment by cyansprite** [comment]
Seems to be fixed by 1.21.1, so no longer an issue, but I would update the versioning in the samples for the sdk so others don't hit this issue

**Comment by az-iot-builder-01** [comment]
@cyansprite, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
